### PR TITLE
[Phoenix] Release 3: Revamp settings and onboarding

### LINKED ISSUE #1569: [Phoenix] New Pomodoro configuration
Allow the user to set their own

- Focus duration (already exists)
- Short-break duration
- Long-break duration
- Take a long break after every Xth break
- No break mode [Low Priority]

### LINKED ISSUE #1570: [Phoenix] Update post-login page
- Remove the close page option button
- Add some gifs for onboarding the user 
 - On how to start/stop a entry
 - Use an integration
 - New Pomodoro
 - Enable integration [low-priority]

### LINKED ISSUE #1571: [Phoenix] Revamp settings page UI to be more webapp
- Link to designs [https://app.zeplin.io/project/58d91e7333cc4a43c790d340/screen/5d653a89a649819b9caa18ff](https://app.zeplin.io/project/58d91e7333cc4a43c790d340/screen/5d653a89a649819b9caa18ff)

### PR DESCRIPTION
**Pull request by shantanuraj**
Please remember the [Contributing Guidelines](https://github.com/toggl/toggl-button/blob/master/.github/CONTRIBUTING.md) :heart:

## :star2: What does this PR do?

- Contains batch of change for 3rd phoenix release
- Including revamped settings page
- Improved onboarding experience
- Improved time entries list style to match the macOS app

## :bug: Recommendations for testing

All changes should be tested across Chrome and Firefox.

Follow recommendations from 

- #1595 - Revamp settings page
- #1598 - Onboarding improvements

## :memo: Links to relevant issues or information

Closes #1571 
Closes #1570

### CODE REVIEW

**Comment by shantanuraj** [comment]
Rebased on latest master

File: src/html/settings.html [hunk 1]
```diff
@@ -4,350 +4,414 @@
...
-              <div class="field__label">
-                <label for="start_automatically">Start timer automatically</label>
-                <div class="description">Starts the timer with the most recent task when the browser is opened</div>
-              </div>
-              <div>
-                <input type="checkbox" id="start_automatically">
-              </div>
+            <span>Pomodoro</span>
+          </li>
+          <li class="nav-link" data-tab="integrations">
```
Comment (tcrammond) [inline src/html/settings.html]: nitpicking but integrations should be higher than pomodoro 😅 it's the most important thing, really. I wonder if we should have a hint banner inside the `General` settings page that if you want to enable Toggl Button in tools, go there.

**Comment by lise-toggl** [comment]
@tcrammond thanks for the thorough review :hugs: 

- TE list hovering issue - could you take a snip of what's happening on your side?
- Settings
  - Remember tab - great idea, I'll add that
  - Focus mode - same
  - Styles - will fix and test on Mac this time too. The filter clear button has had a few issues now so I'll try a different placement method :woman_facepalming: 
  - Support link - you do get a chat box on the user guide (in intercom) but happy to add a more direct one if you think so?
- Onboarding
  - I would love to delete the popup :hocho: 
  - Autostart - if there's extra time before the release I'll work on it :)

File: src/scripts/components/QuickStartGuide.tsx [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        <div css={[section, right, salmon]}>
+          <div>
+            <div>
+              <h5>Easy & fast</h5>
+              <h3>Create time entries</h3>
+              <p>Just click the "Play" icon to start recording your first time entry!</p>
+              <p>When you're done with your task, click on the "Stop" icon to save your entry.</p>
+              <p>The Toggl Button popup also shows a list of all your recent time entries*.</p>
+              <p>You can also enable useful time tracking features like idle detection and reminders from the Settings page.</p>
+              <Link href="settings.html"><Button>Settings</Button></Link>
```
[the remark this answers is not in the document]
Comment (lise-toggl) [reply src/scripts/components/QuickStartGuide.tsx]: The Toggl home page has a little arrow on these buttons, do you think that would be enough?

image

File: src/html/settings.html [hunk 3]
```diff
@@ -4,350 +4,414 @@
...
+              <span class="description">All Toggl Button settings will be reset and you will be logged out.</span>
+            </label>
+            <button id="reset-all-settings" class="danger">Reset all settings to default</button>
+          </div>
+        </section>
+      </div>
+      <div class="tab" data-tab="pomodoro">
+        <section>
+          <div class="headline">
+            <h3>Pomodoro settings</h3>
```
[the remark this answers is not in the document]
Comment (lise-toggl) [reply src/html/settings.html]: some potential links:

- [Wikipedia](https://en.wikipedia.org/wiki/Pomodoro_Technique)
- [founder's website](https://francescocirillo.com/pages/pomodoro-technique)
- [cute video by Trello](https://www.youtube.com/watch?v=ykMzzZolhdk)

we could definitely add a "What is Pomodoro?" expandable box on the page :+1:

File: src/html/settings.html [hunk 4]
```diff
@@ -4,350 +4,414 @@
...
+          <div class="permissions-toolbar" data-id="regular">
+            <a href="javascript:void(0)" id="filter-clear">&times;</a>
+            <input type="text" id="permission-filter" placeholder="Filter integrations"/>
             <div class="button-container">
-                <div id="origins-container"></div>
-                <button id="add-permission">Add</button>
-            </div>
-            <div class="permissions-toolbar" data-id="custom">
+              <button class="enable-all button">Enable all</button>
+              <button class="disable-all button">Disable all</button>
```
[the remark this answers is not in the document]
Comment (lise-toggl) [reply src/html/settings.html]: will add to general style fixes PR :+1:

File: src/html/settings.html (hunk 1 above)
Comment (lise-toggl) [reply src/html/settings.html]: sure, will think about how to add it to `General` :thinking:

**Comment by tcrammond** [comment]
> TE list hovering issue - could you take a snip of what's happening on your side

I'm not sure if it's just a local dev thing (we've had random things before) 🤷‍♂ 

Kapture 2019-12-05 at 11 25 15

File: src/scripts/components/QuickStartGuide.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (lise-toggl) [reply src/scripts/components/QuickStartGuide.tsx]: #1614

File: src/html/settings.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (lise-toggl) [reply src/html/settings.html]: #1614

File: src/html/settings.html (hunk 1 above)
Comment (lise-toggl) [reply src/html/settings.html]: #1614

**Comment by lise-toggl** [comment]
I couldn't repro that at all :( @shantanuraj (when you're back next week) any ideas?

**Comment by lise-toggl** [comment]
rebased on current master

**Pull request merged**